<a href="https://colab.research.google.com/github/JosephHobbs9292/Dojo/blob/main/Project_1_Part_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Joseph Cobbs
--- 

In [1]:
## import pandas and mount drive
from google.colab import drive
drive.mount('/content/drive')

import pandas as pd

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## Loading Data 
---

In [2]:
## Load in dataset
df_sales = pd.read_csv("/content/sales_predictions.csv")
df_sales.info

<bound method DataFrame.info of      Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0              FDA15        9.300          Low Fat         0.016047   
1              DRC01        5.920          Regular         0.019278   
2              FDN15       17.500          Low Fat         0.016760   
3              FDX07       19.200          Regular         0.000000   
4              NCD19        8.930          Low Fat         0.000000   
...              ...          ...              ...              ...   
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
0                     Dairy  249.8092            OUT049 

In [3]:
df_sales.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

## Data Cleaning
---

In [4]:
df_sales.shape


(8523, 12)

In [5]:
df_sales.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

In [6]:
df_sales.duplicated().sum()

0

In [7]:
df_sales.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

There are several options to handle missing data: deleteing the affected columns or row, creating new categories to replace missing values, or simply filling in the  missing data. Since the missing data accounts 1463 values in Item_Weight and 2410 values in Outlet_Size deleteing these rows or columns would lead too great of a data loss. Thus, I will replacing the data with the mean and median values respectively: 


In [8]:
average = df_sales["Item_Weight"].mean()
print(average)

12.857645184135976


In [9]:
median = df_sales["Outlet_Size"].value_counts().median
print(median)

<bound method NDFrame._add_numeric_operations.<locals>.median of Medium    2793
Small     2388
High       932
Name: Outlet_Size, dtype: int64>


In [10]:
df_sales["Item_Weight"].fillna(average, inplace = True)
df_sales["Item_Weight"].isna().sum()

0

In [11]:
df_sales["Outlet_Size"].astype(str)
df_sales["Outlet_Size"].fillna("Medium", inplace = True)
df_sales["Outlet_Size"].isna().sum()

0

## Verifying that all columns have consistent categories
---

In [12]:
df_sales["Item_Fat_Content"].value_counts()

Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: Item_Fat_Content, dtype: int64

In [13]:
df_sales.replace(["LF", "reg", "low fat"], ["Low Fat", "Regular", "Low Fat"], inplace = True)

In [14]:
df_sales["Item_Fat_Content"].value_counts()

Low Fat    5517
Regular    3006
Name: Item_Fat_Content, dtype: int64

In [15]:
df_sales["Item_Type"].value_counts()

Fruits and Vegetables    1232
Snack Foods              1200
Household                 910
Frozen Foods              856
Dairy                     682
Canned                    649
Baking Goods              648
Health and Hygiene        520
Soft Drinks               445
Meat                      425
Breads                    251
Hard Drinks               214
Others                    169
Starchy Foods             148
Breakfast                 110
Seafood                    64
Name: Item_Type, dtype: int64

In [16]:
df_sales["Outlet_Identifier"].value_counts()

OUT027    935
OUT013    932
OUT049    930
OUT046    930
OUT035    930
OUT045    929
OUT018    928
OUT017    926
OUT010    555
OUT019    528
Name: Outlet_Identifier, dtype: int64

In [17]:
df_sales["Outlet_Location_Type"].value_counts()

Tier 3    3350
Tier 2    2785
Tier 1    2388
Name: Outlet_Location_Type, dtype: int64

In [18]:
df_sales["Outlet_Type"].value_counts()

Supermarket Type1    5577
Grocery Store        1083
Supermarket Type3     935
Supermarket Type2     928
Name: Outlet_Type, dtype: int64

## Displaying descriptive statistics of numerical columns
---

In [24]:
## Item_Weight
Item_Weight_Mean = df_sales["Item_Weight"].mean()
Item_Weight_Max = df_sales["Item_Weight"].max()
Item_Weight_Min = df_sales["Item_Weight"].min()
print(f"The Mean is: {Item_Weight_Mean}\nThe Max is: {Item_Weight_Max}\nThe Min is: {Item_Weight_Min}")

The Mean is: 12.857645184135976
The Max is: 21.35
The Min is: 4.555


In [26]:
Item_Visibility_Mean = df_sales["Item_Visibility"].mean()
Item_Visibility_Max = df_sales["Item_Visibility"].max()
Item_Visibility_Min = df_sales["Item_Visibility"].min()
print(f"The Mean is: {Item_Visibility_Mean}\nThe Max is: {Item_Visibility_Max}\nThe Min is: {Item_Visibility_Min}")

The Mean is: 0.06613202877895108
The Max is: 0.328390948
The Min is: 0.0


In [27]:
Item_MRP_Mean = df_sales["Item_MRP"].mean()
Item_MRP_Max = df_sales["Item_MRP"].max()
Item_MRP_Min = df_sales["Item_MRP"].min()
print(f"The Mean is: {Item_MRP_Mean}\nThe Max is: {Item_MRP_Max}\nThe Min is: {Item_MRP_Min}")

The Mean is: 140.9927819781767
The Max is: 266.8884
The Min is: 31.29


In [29]:
Outlet_Establishment_Year_Mean = df_sales["Outlet_Establishment_Year"].mean()
Outlet_Establishment_Year_Max = df_sales["Outlet_Establishment_Year"].max()
Outlet_Establishment_Year_Min = df_sales["Outlet_Establishment_Year"].min()
print(f"The Mean is: {Outlet_Establishment_Year_Mean}\nThe Max is: {Outlet_Establishment_Year_Max}\nThe Min is: {Outlet_Establishment_Year_Min}")

The Mean is: 1997.8318667135984
The Max is: 2009
The Min is: 1985


In [30]:
Item_Outlet_Sales_Mean = df_sales["Item_Outlet_Sales"].mean()
Item_Outlet_Sales_Max = df_sales["Item_Outlet_Sales"].max()
Item_Outlet_Sales_Min = df_sales["Item_Outlet_Sales"].min()
print(f"The Mean is: {Item_Outlet_Sales_Mean}\nThe Max is: {Item_Outlet_Sales_Max }\nThe Min is: {Item_Outlet_Sales_Min}")

The Mean is: 2181.288913575032
The Max is: 13086.9648
The Min is: 33.29
